### ISSUE TITLE
Incomplete fix: do not use nav or navigation role in Tabs component

### ISSUE THREAD

**Issue by carmacleod** [carbon-design-system/carbon#6931]
The Tabs component should not use `<nav>` or `role="navigation"`.

This problem has been fixed before, in 2 separate PRs:
1. https://github.com/carbon-design-system/carbon-components-react/pull/2318
2. https://github.com/carbon-design-system/carbon/pull/2853

And yet, somehow, code is still being generated that wraps the Carbon tablist in `role="navigation"`:
- https://github.com/carbon-design-system/carbon/issues/6124
- https://github.com/carbon-design-system/carbon/issues/5403

This gets people into trouble when they do accessibility testing, because the navigation is invariably flagged with "navigation landmark needs a label", and then the dev adds an `aria-label` to get rid of the violation, but that's the wrong fix. The right fix is to remove the navigation role. This reduces screen reader verbosity when a user focuses the Tabs component. It also reduces the number of navigation landmarks, so that the page doesn't unintentionally end up with too many landmarks.

Carbon components should not be adding landmarks at all. Landmark elements/roles need to be a part of the design of the entire page, and not tied to a component. Landmark elements are: HTML header (in body scope), footer (in body scope), main, nav, aside, section (with accessible name), form (with accessible name). Landmark roles are: banner, contentinfo, main, navigation, complimentary, region (with accessible name), form (with accessible name), search. (It's probably ok for Carbon to define a search landmark for a searchbox, although should probably make sure that it can be overridden with "no role" in case the page design wants the searchbox to be part of a larger search landmark, and not a landmark on its own).

Back to this issue, here are some problematic lines of code:
- [Tabs.js#L45](https://github.com/carbon-design-system/carbon-components-react/blob/master/src/components/Tabs/Tabs.js#L45)
I don't think `role` should be a prop at all on a Tabs component. If an author wants to wrap a Tabs in a `<nav>`, that's up to them, but it needs to be a conscious decision. Please don't give them a way to accidentally do it.

- [Tabs.js#L90](https://github.com/carbon-design-system/carbon-components-react/blob/master/src/components/Tabs/Tabs.js#L90)
Please delete this line.

- [Tabs.js#L31](https://github.com/carbon-design-system/carbon-components-react/blob/master/src/components/Tabs/Tabs.js#L31)
Please delete `<nav>`  from this comment.

There may be other places where the concept of `<nav>` and/or `role="navigation"` needs to be removed from doc, example code, real code, etc. For example, the Storybook for Tabs (snapshot below) uses a navigation role - this should be deleted, too.

![image](https://user-images.githubusercontent.com/3331913/94564858-7c30e200-0236-11eb-9c62-ff3b15764252.png)

**Comment by carmacleod** [comment]
Couple more notes:
- [The APG pattern for a Tabs component](https://w3c.github.io/aria-practices/#tabpanel) does not use nav or navigation role
- [Bootstrap Navs doc](https://getbootstrap.com/docs/4.0/components/navs/#regarding-accessibility) says not to use tab/tablist/tabpanel for navigation (which also means don't use navigation for tab/tablist/tabpanel)

**Comment by emyarod** [comment]
thank you for reporting! I will hold off on making these changes until #6798 is resolved to avoid changes being overlooked or conflicts in merging the fixes

**Comment by carmacleod** [comment]
Ok, that makes sense. Until this is fixed, apparently setting the Tabs component's role to "none" is a (hacky but acceptable) work-around. e.g.

```
<Tabs role="none">
      <Tab label="Tab one">Tab content one</Tab>
      <Tab label="Tab two">Tab content two</Tab>
</Tabs>
```
